### ISSUE TITLE
Ensure all vocabularies have abbreviations and that abbreviations are unique

### ISSUE THREAD

**Issue by dbirman** [AllenNeuralDynamics/aind-data-schema-models#105]
# User story
As a user I want to be able to use the abbreviation field to discriminate vocabulary types

# Acceptance criteria
- [ ] For local vocabularies, ensure all abbreviations are unique and all models have abbreviations
- [ ] For local vocabularies with external validation, ensure all abbreviations are unique and all models have abbreviations
- [ ] For external vocabularies we will use abbreviations when we are sure they are unique

# Sprint Ready Checklist 
- [ ] 1. Acceptance criteria defined 
- [ ] 2. Team understands acceptance criteria 
- [ ] 3. Team has defined solution / steps to satisfy acceptance criteria 
- [ ] 4. Acceptance criteria is verifiable / testable 
- [ ] 5. External / 3rd Party dependencies identified
- [ ] 6. Ticket is prioritized and sized

# Notes
I don't think there's any way to guarantee that external vocabularies map to unique abbreviations in the situation where the external library is very large and we don't have a list of the names/abbreviations available for codegen. I think in that situation we just don't have an abbreviation field at all to avoid users trying to use them in this way.

**Comment by saskiad** [comment]
is the plan for abbreviations to match the name fields when there is no obvious abbreviation? (for local vocab)
Are you proposing that external vocabs have no abbreviations ever, or only if there isn't one provided that we trust?

**Comment by dbirman** [comment]
> is the plan for abbreviations to match the name fields when there is no obvious abbreviation? (for local vocab) Are you proposing that external vocabs have no abbreviations ever, or only if there isn't one provided that we trust?

Yes and the second option, only when we can't trust the external vocab. I'll update the ticket

**Comment by dbirman** [comment]
This adds data in places it wasn't there before, I'm not sure we can guarantee this isn't breaking (adding label)

**Comment by dbirman** [comment]
This doesn't seem important in the end
